src/std/sugar.ss

```diff
@@ -27,7 +27,7 @@
 
 (defrules defrule ()
   ((_ (name args ...) body ...)
-   (defrules name () ((_ args ...) body ...))))
+   (defrules name () ((name args ...) body ...))))
```

Comment:
leave this as `_`, it doesn't make sense to bind the name in the pattern.

Reply:
Yes it does: to pass the name around as a context to use with my `with-id` macro, or any other that breaks hygiene in a controled way.

Reply:
Example use with this patch:
```
(import :std/sugar :clan/with-id)
(def a 1)
(defrule (foo) (with-id foo (a) a))
[(foo) (let (a 2) (foo))]
⇒ [1 2]
```
but that won't work if I can't get to `foo`; I'd get `[1 1]` instead.